apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/TradeWidgetForm.tsx

```diff
@@ -90,6 +105,18 @@ export function TradeWidgetForm(props: TradeWidgetProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
+  const location = useLocation()
+  const isSwapMode = location.pathname.includes('/swap/')
+  const toggleAccountModal = useToggleAccountModal()
+
+  const handleClick = () => {
```

Comment:
```
const handleClick = useCallback(() => {
    if (isSwapMode) {
      toggleAccountModal()
    } else {
      scrollToMyOrders()
    }
  }, [isSwapMode, toggleAccountModal])
```

Reply:
Thank you for the suggestion. Addressed now.